### ISSUE TITLE
Stroke drawing regression in 1.0

### ISSUE THREAD

**Issue by adrian17** [nical/lyon#851]
Noticed in https://github.com/ruffle-rs/ruffle .

Regression was found in commit https://github.com/nical/lyon/commit/5a452b1cb5312041e3a911725a0757ba5e91769a .

Given SVG shape (original is an SWF, so this SVG is a result of a tool-made dump):
```svg
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg xmlns:xlink="http://www.w3.org/1999/xlink" height="334.0px" width="286.2px" xmlns="http://www.w3.org/2000/svg">
  <g transform="matrix(1.0, 0.0, 0.0, 1.0, 20.8, 0.5)">
    <path d="M28.7 284.0 Q33.45 279.2 60.2 269.8 88.0 260.0 99.7 260.0 113.65 260.0 129.7 265.95 L141.3 270.65 Q147.45 273.2 150.55 274.0"
    fill="none" stroke="#000033" stroke-linecap="round" stroke-linejoin="miter-clip" stroke-miterlimit="1.0" stroke-width="14.0"/>
  </g>
</svg>
```

Ruffle draws this:

![2023-01-28_22-53-09](https://user-images.githubusercontent.com/4729533/215292882-45ae5074-7a9e-4f8c-ba17-eac2c564bdf1.png)

Expected: 1 commit earlier, I correctly get this:

![image](https://user-images.githubusercontent.com/4729533/215292868-14359610-12d7-4bb3-b99e-9730ba19a0c2.png)

If you want, I can give you the exact builder calls we do, but this will take a bit more time to generate.

**Comment by nical** [comment]
Hi, thanks for filing this. Unfortunately I couldn't reproduce just from the given path after testing a number of line widths and tolerance parameters.
Having the transformed path, line width and tolerance would help a great deal.

**Comment by adrian17** [comment]
Sure, I'll try to make a better repro soon.

**Comment by adrian17** [comment]
OK, got a repro (with a slightly simpler shape than the one above).
```rs
use lyon::tessellation::{
	LineCap, LineJoin,
	math::Point,
	geometry_builder::{BuffersBuilder, VertexBuffers, Positions},
	StrokeTessellator, StrokeOptions
};

fn main() {
	let stroke_options = StrokeOptions::default()
		.with_line_width(14.0)
		.with_miter_limit(1.0)
		.with_start_cap(LineCap::Round)
		.with_end_cap(LineCap::Round)
		.with_line_join(LineJoin::MiterClip);

	let mut builder = lyon::path::Path::builder();
	builder.begin((28.7, 284.0).into());
	builder.quadratic_bezier_to((33.45, 279.2).into(), (60.2, 269.8).into());
	builder.quadratic_bezier_to((88.0, 260.0).into(), (99.7, 260.0).into());
	builder.end(false);
	let lyon_path = builder.build();

	let mut lyon_mesh: VertexBuffers<Point, u32> = VertexBuffers::new();
	let mut buffers_builder = BuffersBuilder::new(&mut lyon_mesh, Positions{});

	let mut stroke_tess = StrokeTessellator::new();
	stroke_tess.tessellate_path(&lyon_path, &stroke_options, &mut buffers_builder).unwrap();

	println!("{:?}", lyon_mesh);
}
```

On commit 5a452b1cb5312041e3a911725a0757ba5e91769a, I get
```
VertexBuffers { vertices: [(35.03889, 287.9526), (27.366095, 276.19077), (39.424557, 285.55945), (33.26337, 272.97278), (48.092155, 281.7829), (42.847603, 268.7968), (-109.49782, 325.57538), (227.5842, 207.41798), (62.516018, 276.40717), (77.16373, 271.59296), (73.15959, 258.17224), (87.78456, 268.74088), (84.67302, 255.08043), (95.091515, 267.36917), (93.265724, 253.46738), (100.22877, 266.98), (99.17123, 253.02), (106.68, 259.47122), (105.009514, 264.5617), (102.859665, 266.2463), (106.351036, 262.1826), (104.26169, 254.69049), (105.94632, 256.84033), (101.88259, 253.34897), (32.972595, 289.5448), (24.427406, 278.4552), (23.15519, 288.27258), (21.758125, 283.0998), (22.631031, 280.51178), (21.942053, 285.82486), (27.799799, 290.94186), (25.211784, 290.06897), (30.524862, 290.75793)], indices: [1, 0, 2, 1, 2, 3, 3, 2, 4, 3, 4, 5, 5, 4, 8, 5, 8, 6, 6, 8, 7, 7, 8, 9, 7, 9, 10, 10, 9, 11, 10, 11, 12, 12, 11, 13, 12, 13, 14, 14, 13, 15, 14, 15, 16, 15, 17, 16, 15, 18, 17, 15, 19, 18, 18, 20, 17, 17, 21, 16, 17, 22, 21, 21, 23, 16, 25, 24, 0, 25, 0, 1, 25, 26, 24, 25, 27, 26, 25, 28, 27, 27, 29, 26, 26, 30, 24, 26, 31, 30, 30, 32, 24] }
```
In particular, note the huge X coordinates `(-109.49782, 325.57538), (227.5842, 207.41798)`

On the commit before that, I get 
```
VertexBuffers { vertices: [(35.03889, 287.9526), (27.366095, 276.19077), (39.424557, 285.55945), (33.26337, 272.97278), (48.092155, 281.7829), (42.847603, 268.7968), (62.516018, 276.40717), (57.883984, 263.1928), (77.16373, 271.59296), (73.15959, 258.17224), (87.78456, 268.74088), (84.67302, 255.08043), (95.091515, 267.36917), (93.265724, 253.46738), (99.17123, 253.02), (100.22877, 266.98), (106.68, 259.47122), (105.009514, 264.5617), (102.859665, 266.2463), (106.351036, 262.1826), (104.26169, 254.69049), (105.94632, 256.84033), (101.88259, 253.34897), (32.972595, 289.5448), (24.427406, 278.4552), (23.15519, 288.27258), (21.758125, 283.0998), (22.631035, 280.51178), (21.94205, 285.82486), (27.799803, 290.94186), (25.211784, 290.06897), (30.524868, 290.75793)], indices: [1, 0, 2, 1, 2, 3, 3, 2, 4, 3, 4, 5, 5, 4, 6, 5, 6, 7, 7, 6, 8, 7, 8, 9, 9, 8, 10, 9, 10, 11, 11, 10, 12, 11, 12, 13, 13, 12, 15, 13, 15, 14, 15, 16, 14, 15, 17, 16, 15, 18, 17, 17, 19, 16, 16, 20, 14, 16, 21, 20, 20, 22, 14, 24, 25, 23, 24, 26, 25, 24, 27, 26, 26, 28, 25, 25, 29, 23, 25, 30, 29, 29, 31, 23, 24, 23, 1, 23, 0, 1] }
```

**Comment by nical** [comment]
Fix up in lyon_tessellation 1.0.9.
